model/jpa/src/main/java/org/keycloak/connections/jpa/PersistenceExceptionConverter.java

```diff
@@ -75,10 +77,23 @@ private void flushInBatchIfEnabled(Method method) {
         }
     }
 
+    // For JTA, the database operations are executed during the commit phase of a transaction, and DB exceptions can be propagated differently
     public static ModelException convert(Throwable t) {
-        if (t.getCause() != null && t.getCause() instanceof ConstraintViolationException) {
-            throw new ModelDuplicateException(t);
-        } if (t instanceof EntityExistsException || t instanceof ConstraintViolationException) {
+        final Predicate<Throwable> checkDuplicationMessage = throwable -> {
+            final String message = throwable.getCause() != null ? throwable.getCause().getMessage() : throwable.getMessage();
+            return message.toLowerCase().contains("duplicate");
+        };
+
+        Predicate<Throwable> throwModelDuplicateEx = throwable ->
```

Comment:
@ahus1 @vmuzikar 
This commit solves the issue of handling exceptions for the JTA tx type.
You can check the original log[1]. There's only a WARN log where the `org.hibernate.exception.ConstraintViolationException` is thrown, but at the end, we get directly the cause of that (you can see the comment of this method.). 

[1] https://gist.github.com/mabartos/e8ea1e3ea945cf0b22748c40c2ce7e63